std/algorithm/mutation.d

```diff
@@ -540,8 +540,9 @@ See_Also:
         $(LREF uninitializedFill)
         $(LREF initializeAll)
  */
-void fill(Range, Value)(Range range, Value value)
-if (isInputRange!Range && is(typeof(range.front = value)))
+void fill(Range, Value)(auto ref Range range, auto ref Value value)
+if ((isInputRange!Range && is(typeof(range.front = value)) ||
+    isSomeChar!Value && is(typeof(range[] = value))))
```

Comment:
Is the `isSomeChar` nessesary here? Seems redundant if `is(typeof(range[] = value))) == true`